tools/decompile.py

```diff
@@ -1,338 +1,446 @@
+                ),
+                None,
+            )
+            self._overlay = (
+                self.relpath.parts[nonmatching_index - 1] if nonmatching_index else None
+            )
+        return self._overlay
+
+    @property
+    def asm_differ_command(self) -> str:
+        """Constructs the string to run asm-differ if the decompiled function code can be compiled"""
+        return f"{Path(sys.executable).name} {self.root.joinpath("tools/asm-differ/diff.py").relative_to(Path.cwd())} -mwo --overlay {self.overlay} {self.name}"
+
+    @property
+    def context(self):
+        """Leverages m2ctx to create context text"""
+        if not self._context:
+            m2ctx.root_dir = self.root
+            m2ctx.VERSION_DEF = [f"-D_internal_version_{self.version}"]
+            self._context = m2ctx.import_c_file(self.src_path)
+        return self._context
+
+    def decompile(self) -> str:
+        """Leverages m2c to decompile the function and stores it so that decompilation only needs to run once"""
+        if not self.c_code:
+            self.c_code = self._guess_unknown_type(self._run_m2c())
+        return self.c_code
+
+    def upload_to_decompme(self) -> None:
+        """Uploads the decompiled code to decomp.me using the extracted context and SotN decomp.me presets"""
+        print("Loading up the catapult...")
+        # No current expectation of these changing, so hardcoding is probably fine
+        presets = {"us": 14, "pspeu": 132}
+        payload = {
+            "name": self.name,
+            "target_asm": self.asm_code,
+            "context": self.context,
+            "source_code": self.decompile(),
+            "preset": presets[self.version],
+            "diff_label": self.name,
+        }
+
+        try:
+            print("Launching cow...")
+            response = requests.post("https://decomp.me/api/scratch", data=payload)
+            response_json = response.json()
+            if "slug" in response_json:
+                self.scratch_link = (
+                    f"https://decomp.me/scratch/{response_json["slug"]}/"
+                )
+                claim_token = response_json.get("claim_token")
+                if claim_token:
+                    self.scratch_link += f"claim?token={claim_token}"
+                print(
+                    f"{self.name} successfully uploaded to decomp.me and can be {"claimed" if "claim" in self.scratch_link else "accessed"} at {self.scratch_link}"
+                )
+            else:
+                print(
+                    f"Received an error when attempting to upload to decomp.me: {response_json.get("error", response.text)}"
+                )
+        except Exception as e:
+            print(
+                f"An unhandled error occurred while attempting to upload to decomp.me: {e}"
+            )
 
+    def resolve_jumptables(self) -> None:
+        """Adds any jump tables to the function asm file if they aren't already there"""
```

Comment:
I didn't do much to this function since it is mostly deprecated by splat migrating rodata.  Just flattened it down and cleaned it up for readability.